docs/datatypes/concurrency/hub.md

```diff
@@ -203,7 +203,7 @@ The `map` operator allows us to transform the type of messages received from the
 
 The `mapM` operator works the same way except it allows us to perform an effect each time a value is taken from the hub. We could use this for example to log each time a message is taken from the hub.
 
-```scala mdoc:reset:invisible
+```scala mdoc:reset
```

Comment:
Instead of using the `reset` modifier, it is better not to share imports between blocks of codes. To do so, it's better to use `compile-only` modifier for those blocks that are not supposed to be evaluated.

Reply:
Another note would like to mention:
Removing invisible imports isn't sufficient. We should make sure that each code blocks have fresh imports.

This addresses the following issue, which @adamgfraser raised in the https://github.com/zio/zio/issues/4593 issue:
> However, a lot of people are going to copy and paste and compile these examples as written so if there are things missing it makes it harder for them to get started.

Reply:
I want to confirm before I put too much work into wrong path. 

we would like to be EVERY snippet of code be self sufficient when it comes to import? So we can just copy it and run? Should I do anything in a case when snippet is split into two parts with some comment in between when one having both parts compile it? Should both have sufficient imports or only first?

Reply:
We have these types of snippets:
1. Independent snippets: snippets that are not dependent on previous works. They should have all the required imports. 
 - Whenever we do not require evaluating an expression and we don't require showing the result of an expression to the user, I suggest using `compile-only` modifier, which doesn't pollute the scope. The good practice is to use `compile-only` by default.
 - If we need to evaluate the expression, we cannot use the `compile-only` modifier, so after that to clear the previous scopes, we need to use the 'reset' modifier to ensure that the scope is cleared and the next blocks are independent of previous scopes. 
2. Dependent snippets: snippets that depend on previous blocks of code. When a user copies and pastes all those separate blocks, the final code should be compiled, requiring no additional imports. After finishing these groups of related snippets, we should clear the scope using the `reset` modifier.